common/templates/components/task-list/template.njk

```diff
@@ -0,0 +1,36 @@
+{%- from "moj/components/badge/macro.njk" import mojBadge -%}
+<ol class="moj-task-list {{- ' ' + params.classes if params.classes }}" {% for attribute, value in params.attributes %} {{ attribute }}="{{ value }}"{% endfor %}>
+  {%- for section in params.sections %}
+    <li {% for attribute, value in section.attributes %} {{ attribute }}="{{ value }}"{% endfor %}>
+      <h{{ section.heading.headingLevel | default(2) }} class="moj-task-list__section {{- ' ' + section.heading.classes if section.heading.classes }}" {% for attribute, value in section.heading.attributes %} {{ attribute }}="{{ value }}"{% endfor %}>
+        <span class="moj-task-list__section-number">{{ loop.index }}. </span> {{- section.heading.html | safe if section.heading.html else section.heading.text -}}
+      </h{{ section.heading.headingLevel | default(2) }}>
+      {% if section.heading.description %}
+        <div class="govuk-inset-text"> {{ section.heading.description }}</div>
+      {% endif %}
+      <ul class="govuk-!-padding-left-0 moj-task-list__items">
+        {%- for item in section.items %}
+          <li class="moj-task-list__item" {% for attribute, value in item.attributes %} {{ attribute }}="{{ value }}"{% endfor %}>
+            <div class="moj-task-list__task-name">
+              <h3 class="govuk-heading-s govuk-!-margin-bottom-2">
+                {{ item.text }}
+              </h3>
+              {% if item.href %}
+                <a class="govuk-link govuk-link--no-visited-state" href="{{- item.href -}}">
+                  {{- item.hrefText -}}
+                </a>
+              {% else %}
+                <div class="govuk-secondary-text">{{- item.hrefText -}}</div>
+              {% endif %}
+            </div>
+            {% set badgeColor = 'moj-badge--blue moj-task-list__task-completed' if item.complete else 'moj-badge--grey moj-task-list__task-completed' %}
+            {{ mojBadge({
+              text: item.statusText if item.statusText else 'CANNOT START YET',
+              classes: badgeColor
+            }) }}
+          </li>
+        {% endfor -%}
+      </ul>
+    </li>
+  {% endfor -%}
+</ol>
```

Comment:
How do we test nunjuck macros?

Reply:
I've not seen it done - but I guess there must be a way to call a render function for a macro with a given input and check the output. Sounds hairy. Sounds like a lightning talk in the making. Let's look into it.